docs/website/docs/walkthroughs/deploy-a-pipeline/deploy-with-google-cloud-functions.md

```diff
@@ -0,0 +1,75 @@
+---
+title: Deploy with Google Cloud Functions
+description: How to deploy a pipeline with Google Cloud Functions
+keywords: [how to, deploy a pipeline, Cloud Function]
+---
+
+# Deploy a pipeline with Google Could Fuctions
+
+This guide shows you how to deploy a pipeline using the gcloud shell and `dlt` CLI commands. To deploy a pipeline using this method, you must have a working knowledge of GCP and its associated services, such as cloud functions, cloud source repositories, shell editor, IAM and permissions, and GCP service accounts.  
+
+To deploy a pipeline using the GCP cloud functions, you'll first need to set up an empty repo in Cloud Source Repositories, a service provided by GCP for hosting repositories, or you can clone it to your local machine and then install the Google Cloud SDK on your local machine.
+
+## 1. Setup Pipeline in Google Cloud Repositories
+To deploy the pipeline, we'll use the Google Cloud Source Repositories(first method):
+
+1. Sign in to your GCP account and enable the Cloud Functions API.
+2. To set up the environment, you can follow these steps:
+    - Create an empty repo in Cloud Source Repositories.
+      (Cloud Source repository is a service in GCP for hosting repositories from different platforms).
+        
+    - After making the repo, click Edit repo to open it in a "Shell Editor".
+3. In this guide, we'll demonstrate using the `dlt` [Notion verified source](https://dlthub.com/docs/dlt-ecosystem/verified-sources/notion). However, you can use any verified source or create a custom one to suit your needs.
+4. In the shell editor
+    - Run the following command to initialise the verified source with Notion and create a pipeline with BigQuery as the target.
+    
+      ```bash
+      dlt init notion bigquery
+      ```
+    
+    - After running the command, a new directory will be created with the necessary files and configurations.
+    - Detailed information about initialising a verified source and a pipeline example can be found in the `dlthub` [documentation](https://dlthub.com/docs/dlt-ecosystem/verified-sources/notion).
+5. In the notion_pipeline.py method, delete the `if __name__ == "__main__":` method and create a new Python file called "main.py", which can be configured as follows:
```

Comment:
you do not need to delete it. it should work without touching the original script at all

Reply:
Updated the docs with two methods to declare env variables, kindly check.